src/systems/odometry_publisher/OdometryPublisher.cc

```diff
@@ -138,6 +154,24 @@ void OdometryPublisher::Configure(const Entity &_entity,
     this->dataPtr->robotBaseFrame = _sdf->Get<std::string>("robot_base_frame");
   }
 
+  this->dataPtr->dimensions = 2;
+  if (!_sdf->HasElement("dimensions"))
+  {
+    ignwarn << "OdometryPublisher system plugin missing <dimensions>, "
```

Comment:
Not setting dimensions is a valid use case, so we don't need to bother users with a warning. I think `igndbg` should be enough here.